File: 2026/07-July/11-Saturday/solution.js
var countCompleteComponents = function(n, edges) {
    const graph = Array.from({ length: n }, () => []);

    for (const [u, v] of edges) {
        graph[u].push(v);
        graph[v].push(u);
    }

    const visited = new Array(n).fill(false);
    let answer = 0;

    for (let i = 0; i < n; i++) {
        if (visited[i]) continue;

        const stack = [i];
        visited[i] = true;

        let nodes = 0;
        let degreeSum = 0;

        while (stack.length) {
            const cur = stack.pop();
            nodes++;
            degreeSum += graph[cur].length;

            for (const nxt of graph[cur]) {
                if (!visited[nxt]) {
                    visited[nxt] = true;
                    stack.push(nxt);
                }
            }
        }

        if (degreeSum / 2 === (nodes * (nodes - 1)) / 2)
            answer++;
    }

    return answer;
};

File: 2026/07-July/11-Saturday/solution.cpp
class Solution {
public:
    int countCompleteComponents(int n, vector<vector<int>>& edges) {
        vector<vector<int>> graph(n);

        for (auto &e : edges) {
            graph[e[0]].push_back(e[1]);
            graph[e[1]].push_back(e[0]);
        }

        vector<bool> visited(n, false);
        int answer = 0;

        for (int i = 0; i < n; i++) {
            if (visited[i]) continue;

            stack<int> st;
            st.push(i);
            visited[i] = true;

            int nodes = 0;
            int degreeSum = 0;

            while (!st.empty()) {
                int cur = st.top();
                st.pop();

                nodes++;
                degreeSum += graph[cur].size();

                for (int nxt : graph[cur]) {
                    if (!visited[nxt]) {
                        visited[nxt] = true;
                        st.push(nxt);
                    }
                }
            }

            if (degreeSum / 2 == nodes * (nodes - 1) / 2)
                answer++;
        }

        return answer;
    }
};
